liquibase-standard/src/main/java/liquibase/datatype/core/CharType.java

```diff
@@ -41,13 +41,21 @@ public DatabaseDataType toDatabaseDataType(Database database) {
             type.addAdditionalInformation(getAdditionalInformation());
             return type;
         } else if (database instanceof PostgresDatabase){
-            if ((getParameters() != null) && (getParameters().length == 1) && "2147483647".equals(getParameters()[0]
-                .toString())) {
-                DatabaseDataType type = new DatabaseDataType("CHARACTER");
-                type.addAdditionalInformation("VARYING");
+            final Object[] parameters = getParameters();
+            if ((parameters != null) && (parameters.length == 1)) {
+                // PostgreSQL only supports (n) syntax but not (n CHAR) syntax, so we need to remove CHAR.
+                final String parameter = parameters[0].toString().replaceFirst("(?<=\\d+)\\s*(?i)CHAR$", "");
```

Comment:
@mkarg when I check "(?<=\\d+)\\s*(?i)CHAR$" I receive an error that says "* A quantifier inside a lookbehind makes it non-fixed width"
do you think it may result in any problems?

Reply:
@mkarg I wanted to kindly ask if you could take a moment to review the comment I've left in the pull request. Your input and feedback would be greatly appreciated.

Reply:
Sorry for the delay. I will look into this tomorrow.

Reply:
Sorry for the delay. I checked the regex and cannot reproduce your problem in OpenJDK, as OpenJDK's parser is pretty happy with the quantifier and works perfectly well. Apparently the parser you are using is not compliant with the standardized API found in https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html.

So I do not see a need to change my PR, actually.

Reply:
@mkarg thank you!